python_modules/libraries/dagster-airbyte/dagster_airbyte/asset_defs.py

```diff
@@ -88,3 +93,209 @@ def _assets(context):
+        stream_data (List[Mapping[str, Any]]): Unparsed list of dicts with information about each stream.
+    """
+
+    def parse_stream_tables(
+        self, return_normalization_tables: bool = False
+    ) -> Mapping[str, Set[str]]:
+        """
+        Parses the stream data and returns a mapping, with keys representing destination
+        tables associated with each enabled stream and values representing any affiliated
+        tables created by Airbyte's normalization process, if enabled.
+        """
+
+        tables: Mapping[str, Set[str]] = {}
+
+        enabled_streams = [
+            stream for stream in self.stream_data if stream.get("config", {}).get("selected", False)
+        ]
+
+        for stream in enabled_streams:
+            name = cast(str, stream.get("stream", {}).get("name"))
+            prefixed_name = f"{self.stream_prefix}{name}"
+
+            tables[prefixed_name] = set()
+            if self.has_basic_normalization and return_normalization_tables:
+                for k, v in stream["stream"]["json_schema"]["properties"].items():
+                    for normalization_table_name in _get_normalization_tables_for_schema(
+                        k, v, f"{name}_"
+                    ):
+                        prefixed_norm_table_name = f"{self.stream_prefix}{normalization_table_name}"
+                        tables[prefixed_name].add(prefixed_norm_table_name)
+
+        return tables
+
+
+class AirbyteSource(NamedTuple("_AirbyteSource", [("name", str)])):
+    """
+    Contains information about an Airbyte source.
+
+    Attributes:
+        name (str): The name of the source.
+    """
+
+
+def _airbyte_connection_from_config(contents: Mapping[str, Any]) -> AirbyteConnection:
+    config_contents = cast(Mapping[str, Any], contents.get("configuration"))
+    check.invariant(
+        config_contents is not None, "Airbyte connection config is missing 'configuration' key"
+    )
+
+    return AirbyteConnection(
+        name=contents["resource_name"],
+        source_config_path=contents["source_configuration_path"],
+        stream_prefix=config_contents.get("prefix", ""),
+        has_basic_normalization=any(
+            op.get("operator_configuration", {}).get("operator_type") == "normalization"
+            for op in config_contents.get("operations", [])
+        ),
+        stream_data=config_contents.get("sync_catalog", {}).get("streams", []),
+    )
+
+
+def _airbyte_source_from_config(contents: Mapping[str, Any]) -> AirbyteSource:
+    return AirbyteSource(name=contents["resource_name"])
+
+
+@experimental
+def load_assets_from_airbyte_project(
+    project_dir: str,
+    key_prefix: Optional[CoercibleToAssetKeyPrefix] = None,
+    source_key_prefix: Optional[CoercibleToAssetKeyPrefix] = None,
+    create_assets_for_normalization_tables: bool = True,
+    connection_to_group_fn: Optional[Callable[[str], Optional[str]]] = _clean_name,
+) -> Tuple[List[AssetsDefinition], List[SourceAsset]]:
```

Comment:
Returning a `List[Union[AssetsDefinition, SourceAsset]]` would be more consistent with functions like `load_assets_from_package_module`.

Reply:
Makes sense - how do you pull it back out into assets and source assets for e.g. `build_assets_job` then?

Reply:
Users don't call `build_assets_job` - normally they'd either call `materialize` or put them on their repo, both of which don't require separating them out.